test/uzal_cost_test.jl

```diff
@@ -0,0 +1,201 @@
+using DynamicalSystemsBase
```

Comment:
This test file needs to be reworked into proper unit tests.

1. It cannot be using plotting packages (but the plotting code can remain, but it has to be commented out)
2. There is no reason to save and load text files, instead `DynamicalSystemsBase` can be used to produce the timeseries on demand.
3. Unit tests are performed as follows: use the `@test` macro and try to test the the returned `L` number matches something expected, either by theory or by existing literature. If you have a look at the rest of the test files it will help you structure the tests accordingly.
4. The test file can probably be shortened, at the moment it is about 200 lines but I think we can do it in 50 or so.

Reply:
Ah, also the text file `roess_ts.txt` has to be deleted.

Reply:
Thanks for the comments. `roess_ts.txt` was added just for comparison reasons to my Matlab implementation. Regarding the expectation value of the L-statistic: I don't see a way of an independent test, i.e. an analytical reference result. The only thing I know are rough values from the Paper itself and standardized outputs from this implementation (and from my Matlab implementation).

Reply:
That is fine, if this is the best we have this is what we do! You can use the following syntax: `@test Lmin < L < Lmax` for some pre-defined, empirical minimum and maximum values.

Perhaps you also want to run it with a vector of ones, `v = ones(10000)` and see what it does.

Reply:
Yah, I will do. This is also related to your comment below: The total value of `L` depends on the time series itself (i.e. a normalization will yield different values, but does not change the qualitative curve, when plotted against `t_w`, for instance. It also slightly depends on the prediciton horizon `Tw`. So they did not comment on this in Fig.7 in the paper. That is why I am not really concerned about the different scale. Another thing is the assertion of a column vector Dataset input. I could of course assert the input as an `AbstractDataset`-type, but how can I assert, that this is a column-vector then (execpt from manually checking the `size` and possibly transpose)?

Reply:
Actually, I need to refine my last comment. The scale itself is very important, when comparing the minima of course. The rank of the minima, with respect to the total (negative) value, must be preserved, i.e. the same as shown in the Paper. So I think/hope, it is all good :)

Reply:
Okay this is convincing for me. 

> Another thing is the assertion of a column vector Dataset input. I could of course assert the input as an AbstractDataset-type, but how can I assert, that this is a column-vector then (execpt from manually checking the size and possibly transpose)?

I did not understand this part... can you please try to explain it differently? WHat I can say for sure is that manually checking anything and transposing anything should definitely be avoided.

Reply:
The function works properly, when the input phase space trajectory `Y` is a Dataset "filled" with column-vectors. Meaning, the number of columns represent the dimensionality of the phase space trajectory and the number of rows the number of observations. I somehow need to assert, that `Y`is really that kind of Dataset. So I could assert this, as far as I understand within the definition of the function, typing `Y::AbstractDataset`. But then it is still not guaranteed, that it is filled with column-vectors. - So one would need to check this within the function body. Moreover, a normal Julia-`Array`, would also be fine, then one would need to convert it into a `Dataset` within the function body. Maybe there is an elegant way to assert this?

Reply:
The `AbstractDataset` type guarantees that you have column number = dimensionality. The assertion is enough.

Reply:
`Dataset` is not a `Matrix`, internally it is a very different thing.

Reply:
Perfect!

Reply:
but then you have to change the documentation string, because it says that it can get as an input either timeseries or Dataset, while in reality it can only get datasets.